demos/tutorials/RandomVariables.py

```diff
@@ -0,0 +1,384 @@
+# random variables (e.g. the covariance of y is lazily evaluated as 1/s).
+
+d = Gamma(1, 1e-4).rv
+s = Gamma(1, 1e-4).rv
+x = LMRF(0, 1/d, geometry=A.domain_geometry).rv
+y = Gaussian(A @ x, 1/s).rv
+
+# %%
+######################################################################
+# For illustrative purposes let us put this into a Bayesian Problem
+# object:
+# 
+
+BP = BayesianProblem(d, s, x, y)
+print(BP)
+
+# %%
+######################################################################
+# Providing data to the Bayesian problem creates a posterior as shown.
+#
+
+BP.set_data(y=y_obs)
+print(BP)
+
+# %%
+######################################################################
+# Given the Bayesian problem posterior we can now sample from
+# the posterior distribution using MCMC as follows:
+#
+
+samples = BP.sample_posterior(1000)
+
+# %%
+######################################################################
+# We can now plot a 95 credibility interval of the samples for x
+# from the posterior distribution as follows:
+#
+
+samples["x"].plot_ci(exact=info.exactSolution)
+
+# %%
+# Revisiting the random variable methods and attributes
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# In this section we revisit the methods and attributes of random variables
+# introduced earlier in this tutorial.
+# 
+# For this purpose let us define a new random 3-dimensional
+# variable :math:`x` as follows:
+
+x = Gaussian(0, 1, geometry=3).rv
+
+# %%
+######################################################################
+# Probability density evaluation
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Probability density evaluation works by evaluating the probability
+# density function of the random variable at a given point. This is done
+# using the ``logd`` method. **Note**. Only simple (non-transformed)
+# random variables support this method at this point in time.
+#
+
+x.logd([1, 2, 3])
+
+# %%
+######################################################################
+# Gradient evaluation
+# ^^^^^^^^^^^^^^^^^^^
+#
+# Gradient evaluation works by evaluating the gradient of the log
+# probability density function of the random variable at a given point.
+#
+
+x.gradient([1, 2, 3])
+
+
+# %%
+######################################################################
+# Dimension and geometry
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+# The dimension of a random variable can be accessed using the ``dim``
+# attribute. The geometry of a random variable can be accessed using the
+# ``geometry`` attribute.
+#
+
+print(x.dim)
+print(x.geometry)
+
+# %%
+######################################################################
+# Conditioning
+# ^^^^^^^^^^^^
+#
+# Conditioning works by conditioning the underlying distribution of the
+# random variable on the given input. This is done using the ``condition``
+# method. For more details see documentation of conditioning on distributions.
+
+z = Gaussian(0, lambda s: s).rv
+
+print(z)
+print(z.condition(s=10))
```

Comment:
Is there a reason to not use the simpler syntax `z(s=10)`, which would be closer to the equivalent for distribution?

Reply:
Yes, the reason is that distributions function function with general "callable" objects in the sense that one can create conditional distributions from using any callable object like a method, a lambda function or an object with the __call__ method implemented.

While we could special case RVs in the distribution implementation, I would rather use this syntax. In fact, we had previously discuseed if we should use a more explicit syntax for conditioning Distributions aswell, rather than the call syntax.

To align RVs and distributions we could thus consider using `.condition` or `.given` as syntax for conditioning. This could be updated in a new PR to avoid keeping this one alive for too long though. See #250